learning_resources/etl/mitpe.py

```diff
@@ -0,0 +1,411 @@
+"""Professional Education ETL"""
+
+import copy
+import json
+import logging
+import re
+from datetime import UTC, datetime
+from hashlib import md5
+from urllib.parse import urljoin
+from zoneinfo import ZoneInfo
+
+import requests
+from django.conf import settings
+
+from learning_resources.constants import (
+    CertificationType,
+    LearningResourceFormat,
+    OfferedBy,
+    PlatformType,
+)
+from learning_resources.etl.constants import ETLSource
+from learning_resources.etl.utils import clean_data, transform_topics
+from main.utils import now_in_utc
+
+log = logging.getLogger(__name__)
+
+BASE_URL = "https://professional.mit.edu/"
+OFFERED_BY = {"code": OfferedBy.mitpe.name}
+UNIQUE_FIELD = "url"
+
+
+def _fetch_data(url, params=None) -> list[dict]:
+    """
+    Fetch data from the Professional Education API
+
+    Args:
+        url(str): The url to fetch data from
+        params(dict): The query parameters to include in the request
+
+    Yields:
+        list[dict]: A list of course or program data
+    """
+    if not params:
+        params = {}
+    while url:
+        response = requests.get(
+            url, params=params, timeout=settings.REQUESTS_TIMEOUT
+        ).json()
+        results = response["data"]
+
+        yield from results
+        url = response.get("links", {}).get("next", {}).get("href")
+
+
+def get_related_object(url: str) -> dict:
+    """
+    Get the related object data for a resource
+
+    Args:
+        url(str): The url to fetch data from
+
+    Yields:
+        dict: JSON data representing a related object
+
+    """
+    response = requests.get(url, timeout=settings.REQUESTS_TIMEOUT).json()
+    return response["data"]
+
+
+def extract() -> list[dict]:
+    """
+    Load the Professional Education data from an external API
+
+    Returns:
+        list[dict]: list of raw course or program data
+    """
+    if settings.PROFESSIONAL_EDUCATION_API_URL:
+        return list(_fetch_data(settings.PROFESSIONAL_EDUCATION_API_URL))
+    else:
+        log.warning("Missing required setting PROFESSIONAL_EDUCATION_API_URL")
+
+    return []
+
+
+def parse_topics(resource_data: dict) -> list[dict]:
+    """
+    Get a list containing {"name": <topic>} dict objects
+
+    Args:
+        resource_data: course or program data
+
+    Returns:
+        list of dict: list containing topic dicts with a name attribute
+    """
+    topic_data = resource_data["relationships"]["field_course_topics"]
```

Comment:
Can this be 
```
topics_exist = resource_data["relationships"]["field_course_topics"]["data"] 
```
instead of reusing the variable name `topic_data ` to mean different things